astrogrism/simulate/simulate.py

```diff
@@ -0,0 +1,86 @@
+from shutil import copy
+from tarfile import open as tar_open
+from tempfile import gettempdir
+from warnings import warn
+
+from astropy.utils.data import download_file
+from synphot import Observation
+
+
+def generate_simulation_spectrum(grism, detector=None, temp_path=gettempdir()):
+    """
+    Initializes and uses STSynphot to generate a Vega spectrum within the bandpass of a given grism
+
+    Parameters
+    ----------
+    grism : str
+        String representation of one of the four supported HST Grisms
+        Valid grisms: G141, G102, G280, G800L
+
+    detector : int
+        For detectors with multiple chips, specifies which chip to simulate
+        Only useful for G280 and G800L Grisms
+
+    temp_path : str
+        Path to download necessary files for STSynphot. Fallsback to Python's
+        default temporary folder location
+
+    """
+    SIM_DATA_DIR = Path(temp_path) / "astrogrism_simulation_files"
+
+    if detector not in (1, 2, None):
+        raise ValueError("Invalid detector argument. Please choose 1 or 2")
+
+    # Prepare environment required for stsynphot to be imported
+    environ['PYSYN_CDBS'] = str(SIM_DATA_DIR / 'grp' / 'redcat' / 'trds')
+
+    # Always download our data files fresh. Ensures changes from redcat team will be picked up
+    # Download HST Instrument Data Files
+    SIM_DATA_DIR.mkdir(parents=True, exist_ok=True)
+    hst_data_files_archive = Path(download_file('https://ssb.stsci.edu/trds/tarfiles/synphot1.tar.gz', cache=True)) # noqa
+    with tar_open(hst_data_files_archive) as tar:
+        tar.extractall(path=SIM_DATA_DIR)
+    # Download Vega CALSPEC Reference Atlas
+    calspec_dir = SIM_DATA_DIR / 'grp' / 'redcat' / 'trds' / 'calspec'
+    calspec_dir.mkdir(parents=True, exist_ok=True)
+    vega_reference_atlas = Path(download_file('https://archive.stsci.edu/hlsps/reference-atlases/cdbs/current_calspec/alpha_lyr_stis_010.fits', cache=True)) # noqa
+    copy(str(vega_reference_atlas), str(calspec_dir / 'alpha_lyr_stis_010.fits'))
+
+    # Now that we have all our reference files, we can import stsynphot
+    # (This is why it's not a top-line import)
+    from stsynphot import Vega, band # noqa
+    if grism == 'G141':
+        if detector:
+            warn("WFC3's G141 grism does not have multiple detectors. Ignoring detector argument",
+                 RuntimeWarning)
+        bandpass = band('wfc3,ir,g141')
+    elif grism == 'G102':
+        if detector:
+            warn("WFC3's G102 grism does not have multiple detectors. Ignoring detector argument",
+                 RuntimeWarning)
+        bandpass = band('wfc3,ir,g102')
+    elif grism == 'G280':
+        bandpass = band(f'wfc3,uvis{detector},g280')
+    elif grism == 'G800L':
+        bandpass = band(f'acs,wfc{detector},g800l')
+    else:
+        raise ValueError(f"Unrecognized grism: {grism}. Valid grisms: G141, G102, G280, G800L")
+
+    spectrum = Observation(Vega, bandpass, binset=bandpass.binset).to_spectrum1d()
```

Comment:
This will work. But if you want the binned flux, we might need to do this differently.

For future reference, API doc: https://synphot.readthedocs.io/en/latest/api/synphot.spectrum.BaseSpectrum.html?highlight=to_spectrum1d#synphot.spectrum.BaseSpectrum.to_spectrum1d

Reply:
So the binning situation is interesting. if we peak forward at https://github.com/spacetelescope/astrogrism/pull/19/files#diff-a44ad3e831c16508917324e007e1c88a870570691f40e4410b6fd5782c37b6a1R98-R101, I'm just "flooring" which "pixel bin" the spectra should disperse into. This is intended as a stand-in until we start working on the next step, that being drizzling. Binning is something we're thinking about very soon, but probably preemptive at this stage. BUT it's nice to expose it here so that we can modify it if we so choose later on

Reply:
If you are interested in how Observation do the binning, maybe start at https://synphot.readthedocs.io/en/latest/synphot/observation.html

And if you want to look at the code, see https://github.com/spacetelescope/synphot_refactor/blob/master/synphot/binning.py and https://github.com/spacetelescope/synphot_refactor/blob/master/synphot/src/synphot_utils.c

Reply:
Definitely good context! The weird thing is we're ultimately trying to bin against the pixels on the final image, so that mapping between the spectral binning and spatial binning will need to be a discussion some point in the near future. These will be useful as we try to figure out exactly what we want to do